jobserver/forms.py

```diff
@@ -115,12 +113,18 @@ def __init__(self, project, repos_with_branches, *args, **kwargs):
 
         # has there been a repo selected already?
         if "data" in kwargs and "repo" in kwargs["data"]:
-            repo = first(
-                self.repos_with_branches,
-                key=lambda r: r["url"] == kwargs["data"]["repo"],
+            repo = next(  # pragma: no branch
+                (
+                    r
+                    for r in self.repos_with_branches
+                    if r["url"] == kwargs["data"]["repo"]
+                ),
+                None,
             )
         else:
-            repo = first(self.repos_with_branches)
+            repo = next(  # pragma: no branch
+                (r for r in self.repos_with_branches if r), None
```

Comment:
From looking at the code above, it doesn't look like we're expecting `r` to be falsy here, so can we just write `repo = self.repos_with_branches[0]` instead?

Reply:
I think this is reasonable.